File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DateTimePicker/Features/MinDatePropertyPage.vb

Namespace CompanionKit.Controls.DateTimePicker.Features

    Public Class MinDatePropertyPage

        Public Sub New()

            ' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()

        End Sub

        ''' <summary>
        ''' Handles ValueChanged event of the DateTimePicker that
        ''' represents allowed minimum date for the demonstrated DateTimePicker.
        ''' </summary>
        Private Sub mobjMinDateTimePicker_ValueChanged(sender As Object, e As EventArgs) Handles mobjMinDateTimePicker.ValueChanged
            Dim mobjDateTimePicker As Gizmox.WebGUI.Forms.DateTimePicker = TryCast(sender, Gizmox.WebGUI.Forms.DateTimePicker)

            ' Change minimum limit of the DateTimePicker.
            Me.mobjDemoDateTimePicker.MinDate = mobjDateTimePicker.Value
        End Sub

        ''' <summary>
        ''' Handles Load event of the example' UserControl.
        ''' </summary>
        Private Sub MinDatePropertyPage_Load(sender As Object, e As EventArgs) Handles Me.Load
            ' Set initial value for the DateTimePicker as minimum date of the demonstrated DateTimePicker
            Me.mobjMinDateTimePicker.Value = Me.mobjDemoDateTimePicker.MinDate
        End Sub
    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DateTimePicker/Features/MinDatePropertyPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.DateTimePicker.Features
{
    public partial class MinDatePropertyPage : UserControl
    {
        public MinDatePropertyPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles ValueChanged event of the DateTimePicker that 
        /// represents allowed minimum date for the demonstrated DateTimePicker.
        /// </summary>
        private void mobjMinDateTimePicker_ValueChanged(object sender, EventArgs e)
        {
            Gizmox.WebGUI.Forms.DateTimePicker mobjDateTimePicker = sender as Gizmox.WebGUI.Forms.DateTimePicker;

            // Change minimum limit of the DateTimePicker.
            this.mobjDemoDateTimePicker.MinDate = mobjDateTimePicker.Value;
        }

        /// <summary>
        /// Handles Load event of the example' UserControl.
        /// </summary>
        private void MinDatePropertyPage_Load(object sender, EventArgs e)
        {
            // Set initial value for the DateTimePicker as minimum date of the demonstrated DateTimePicker
            this.mobjMinDateTimePicker.Value = this.mobjDemoDateTimePicker.MinDate;
        }
    }
}